packages/sdk/app-framework/src/plugins/IntentPlugin/provides.ts

```diff
@@ -21,3 +21,15 @@ export const parseIntentPlugin = (plugin: Plugin) =>
 
 export const parseIntentResolverPlugin = (plugin: Plugin) =>
   (plugin.provides as any).intent?.resolver ? (plugin as Plugin<IntentResolverProvides>) : undefined;
+
+//
+// Intents
+//
+
+const INTENT_ACTION = 'dxos.org/plugin/intent';
+export enum IntentAction {
+  /**
+   * Fired after an intent is dispatched if the intent is undoable.
+   */
+  UNDO_AVAILABLE = `${INTENT_ACTION}/undo-available`,
```

Comment:
I think `SHOW_UNDO` is more consistent. unless there's an even broader intent such as HISTORY_UPDATED where the layout would get the current state of the history (incl. undo being available).